mitodl/mit-open
# Configurable, Tabbed Carousels

### What are the relevant tickets?
https://github.com/mitodl/hq/issues/3940

### Description (What does it do?)
This PR:
- Removes the existing TitledCarousel component, replacing it with simpler `Carousel`.
- Adds a `TabbedCarousel` component. See screenshot. The idea is to create a reusable tab+tabpanel component where each tabpanel has carousel, and the carousel data is pulled from an API based on the configuration.

### Screenshots (if appropriate):
<img width="1526" alt="Screenshot 2024-04-04 at 3 36 27 PM" src="https://github.com/mitodl/mit-open/assets/9010790/be4949e0-6f81-4809-a7bf-33e04872b23c">


### How can this be tested?
1. View the homepage. It should have two "Tabbed Carousels", one for upcoming courses, the other for Media. They should work. (try clicking tabs, clicking the buttons, etc).
    - NB: The "Professional" tab in Upcoming Courses might be empty unless you're using production xpro data.
3. You might like to try changing the tab configuration.
    - In frontends/mit-open/src/pages/HomePage/HomePage.tsx, try changing `UPCOMING_COURSES_CAROUSEL` or `MEDIA_CAROUSEL`.  You can change the `params` objects to change the API filters, or add a new object to get a new tab.
